app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/AddPictureActivity.kt

```diff
@@ -0,0 +1,126 @@
+package com.github.HumanLearning2021.HumanLearningApp
+
+import android.Manifest
+import android.content.pm.PackageManager
+import android.os.Bundle
+import android.view.View
+import android.widget.Button
+import android.widget.Toast
+import androidx.activity.result.ActivityResultLauncher
+import androidx.activity.result.contract.ActivityResultContracts
+import androidx.appcompat.app.AlertDialog
+import androidx.appcompat.app.AppCompatActivity
+import androidx.camera.core.CameraSelector
+import androidx.camera.core.Preview
+import androidx.camera.lifecycle.ProcessCameraProvider
+import androidx.camera.view.PreviewView
+import androidx.core.content.ContextCompat
+import androidx.lifecycle.LifecycleOwner
+import com.google.common.util.concurrent.ListenableFuture
+
+/*
+Activity where an administrator can take a picture to add it to a selected data set
+ */
+class AddPictureActivity : AppCompatActivity() {
+
+    private lateinit var cameraProviderFuture: ListenableFuture<ProcessCameraProvider>
+    private lateinit var requestPermissionLauncher: ActivityResultLauncher<String>
+    private lateinit var openLabelSelectionLayoutListener: View.OnClickListener
+
+    private val dummyLabels = arrayOf("label1", "label2", "label3", "label4", "label5", "label6", "label7", "label8", "label9", "label10")
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setupRequestPermissionLauncher()
+        setupOpenLabelSelectionLayoutListener()
+
+        if (cameraIsAvailable()) {
+            when {
+                ContextCompat.checkSelfPermission(
+                        this,
+                        Manifest.permission.CAMERA
+                ) == PackageManager.PERMISSION_GRANTED -> {
+                    setupCameraPreviewView()
+                }
+                shouldShowRequestPermissionRationale(Manifest.permission.CAMERA) -> {
+                    permissionNeededDialog()
+                }
+                else -> {
+                    requestPermissionLauncher.launch(Manifest.permission.CAMERA)
+                }
+            }
+        } else {
+            permissionNeededDialog()
+        }
+    }
+
+    private fun cameraIsAvailable(): Boolean {
+        return applicationContext.packageManager.hasSystemFeature(PackageManager.FEATURE_CAMERA)
+    }
+
+    private fun setupRequestPermissionLauncher() {
```

Comment:
Would it be possible to override `onRequestPermissionsResult` for that?

Reply:
Yes but as mentioned [here]https://developer.android.com/training/permissions/requesting#request-permission) it would require managing the permission code which is not necessary this way.